atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/pool/CassandraServiceTest.java

```diff
@@ -46,6 +50,22 @@
     private CassandraKeyValueServiceConfig config;
     private Blacklist blacklist;
 
+    @Test
+    public void shouldReturnOnlyLocalHost() {
+        ImmutableSet<InetSocketAddress> hosts = ImmutableSet.of(HOST_1, HOST_2);
+        ImmutableSet<InetSocketAddress> localHosts = ImmutableSet.of(HOST_1);
```

Comment:
So looking at `filterHostsByDatacentre` there's a piece where if there is no intersection you return the original hosts, is there a relevant test you can add there? does it make sense to add this test? is the set of hosts passed in always a subset of `hosts`?